services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/tenant/AutowiredTenantInformationService.java

```diff
@@ -53,6 +55,18 @@
                 });
     }
 
+    @Override
+    public Future<TenantResult<TenantObject>> getTenant(final String tenantId, final Span span) {
+        return service.get(tenantId, span)
+                .flatMap(result -> {
+                    if (result.isOk()) {
+                        return Future.succeededFuture(TenantResult.from(HttpURLConnection.HTTP_OK, result.getPayload().mapTo(TenantObject.class)));
+                    } else {
+                        return Future.failedFuture("cannot get tenant [" + tenantId + "]: " + result.getStatus() + " " + result.getPayload());
```

Comment:
According to the Javadoc in the interface, this method returns a succeeded Future with 404 status if no corresponding tenant was found. So, a `result.isNotFound()` case, returning a succeeded Future, seems to be missing here.

Reply:
I adapted the JavaDoc, since there is no client code relying on this.

Reply:
I think returning a succeeded Future with 404 result would be more consistent here, though. That is also the behaviour of `TenantService.get` and `CredentialsService.get` for example.

Reply:
In my opinion this just adds more unnecessary code which has to be maintained (and understood). I tend to follow the YAGNI principle here rather. If there's actually more than one client, this code can still be adapted.

Reply:
FMPOV, letting the places where `getTenant` gets called also handle the `isError()` case, would also make the code more robust against other `TenantInformationService` implementations which might (by accident) just pass along the `TenantService.get` result.
And passing along that result would probably make the code above simpler. I guess this should work:
`return Future.succeededFuture(result.map(p -> p.mapTo(TenantObject.class)));`  (instead of the "if.. else" block).
@kaniyan @sophokles73 Other opinions?

Reply:
I think the result cannot be mapped to a TenantObject in an error case, since there is no such object in the payload then.

Hence the if clause is still required as far as I can see. It will then have to return null as the tenant object.

Reply:
The `p -> p.mapTo(TenantObject.class)` parameter will only be applied on a non-null payload, the `TenantResult.map()` method using a null TenantObject in the created TenantResult if the payload is null. But your solution works just as well.